Spec/RealtimeClientPresence.swift

```diff
@@ -957,6 +957,94 @@ class RealtimeClientPresence: QuickSpec {
+                        defer { clientSubscribed.dispose(); clientSubscribed.close() }
+                        let channelSubscribed = clientSubscribed.channels.get("foo")
+                        channelSubscribed.attach()
+
+                        let clientPresentMember = ARTRealtime(options: options)
+                        defer { clientPresentMember.dispose(); clientPresentMember.close() }
+                        let channelPresentMember = clientPresentMember.channels.get("foo")
+
+                        var hasInconsistentConnectionIdMethodCalls = 0
+                        let hook = channelSubscribed.presenceMap.testSuite_injectIntoMethodAfter(NSSelectorFromString("hasInconsistentConnectionId")) {
+                            hasInconsistentConnectionIdMethodCalls += 1
+                        }
+                        defer { hook.remove() }
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channelPresentMember.presence.enterClient("tester", data: nil) { error in
+                                expect(error).to(beNil())
+                                done()
+                            }
+                        }
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channelSubscribed.presence.get { presences, error in
+                                expect(error).to(beNil())
+                                expect(presences).to(haveCount(1))
+                                done()
+                            }
+                        }
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channelSubscribed.presence.subscribe(.Leave) { presence in
+                                // Check `synthesized leave` event
+                                expect(presence.id).toNot(equal("\(presence.connectionId):0:0"))
+                                done()
+                            }
+                            clientPresentMember.close()
+                        }
+
+                        expect(channelSubscribed.presenceMap.members).to(beEmpty())
+                        expect(hasInconsistentConnectionIdMethodCalls) == 1
+                    }
+
+                    // RTP2b2
+                    it("split the id of both presence messages") {
+                        let options = AblyTests.commonAppSetup()
+                        let client = ARTRealtime(options: options)
+                        defer { client.dispose(); client.close() }
+                        let channel = client.channels.get("foo")
+
+                        var compareByMsgSerialAndOrIndexMethodCalls = 0
+                        let hook = channel.presenceMap.testSuite_injectIntoMethodAfter(NSSelectorFromString("compareByMsgSerialAndOrIndex")) {
+                            compareByMsgSerialAndOrIndexMethodCalls += 1
+                        }
+                        defer { hook.remove() }
```

Comment:
This test is better than the one above, but it does not confirm that if say the timestamp for a new message is older yet the ID check indicates it's newer (and the connection ID matches the first part of the message), then it works as expected.  I would consider determining the connectionId and injecting messages at a protocol level to test the permutations at https://github.com/ably/ably-ruby/pull/99/commits/e699106b0c55a82040336b4aa3a58a95f620546f#diff-c9dcfc1419f386559fc4a8a0ef368c99R617